wadas/domain/ai_model.py

```diff
@@ -184,17 +231,58 @@ def process_video_offline(self, video_path):
         # Run the detection model on the video frames
         detection_lists = self.detection_pipeline.run_detection(frames, AiModel.detection_threshold)
 
-        # Classify detected animals on all frames
-        classification_lists = self.detection_pipeline.classify(
-            frames, detection_lists, AiModel.classification_threshold
-        )
+        preview_frames = []
+        output_video_path = ""
+        if classification:
+            logger.info("Running classification on video %s ...", video_path)
+            # Classify detected animals on all frames
+            classification_lists = self.detection_pipeline.classify(
+                frames, detection_lists, AiModel.classification_threshold
+            )
+
+            for frame, classified_animals in zip(frames, classification_lists):
+                array = np.array(frame)
+                tracked_animal = tracker.update(classified_animals, array.shape[:2])
+                tracked_animals.append(tracked_animal)
+
+                if save_processed_video:
+                    classified_frame = self.build_classification_square(
+                        frame, classified_animals, "", True
+                    )
+                    if classified_frame:
+                        preview_frames.append(classified_frame)
+                    else:
+                        # If frame does not contain classification keep original frame
+                        # to build output video
+                        preview_frames.append(frame)
```

Comment:
# If frame does not contain classification keep original frame
                    # to build output video
                    preview_frames.append(classified_frame if classified_frame else frame)